grouper/repositories/service_account.py

```diff
@@ -52,8 +54,15 @@ def create_service_account(self, name, owner, machine_set, description):
         user.add(self.session)
         service.add(self.session)
 
-        # Flush the account to allocate an ID, and then create the linkage to the owner.
+        # Flush the account to allocate an ID.
         self.session.flush()
+
+        # Set initial user metadata fields if present.
+        if initial_metadata is not None:
+            for key, value in initial_metadata.items():
+                set_user_metadata(self.session, user.id, key, value)
```

Comment:
This is probably not the "cleanest" solution, and eventually should be cleaned up by adding a `UserMetadata` service and repository later.

Reply:
Same thought; mixing in non-hexagonal code like this is a bit unfortunate. Not the hugest deal in this case I think since the `set_user_metadata` function is pretty clean -- it's pretty much exactly what a `UserMetadata` repo would be doing anyway -- but could you leave a comment or TODO noting it?

Reply:
Ah relatedly, `set_user_metadata` uses a raw assert here to check the key against a regex: https://github.com/dropbox/merou/blob/198a3a9cba56170f3896dd91ee3dddc5f7c11286/grouper/user_metadata.py#L55

Probably worth making that a nicer catchable exception